### PR TITLE
O3-1284: Compact Patient Search v1

### PR DESCRIPTION
**Pull request by vasharma05**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

This PR introduces compact and stable patient search for OpenMRS3. The navigation button for searches now opens a dropdown to select patient from the list.

Key features include:
1. Infinte scrolling, in case of large number of results.
2. Age in the patient banner is in terms of years, months and days.
3. Added extensionSlot with slotname as 'search-patient-button', which when used will render the overlay to search patients. `selectPatientAction` function can be passed to the extension-slot, to use the selected patient from the list, according to your function, without navigating to the patient chart.

## Screenshots
image
image
image
image
image
image
image
image

## Related Issue

https://issues.openmrs.org/browse/O3-1284

## Other

*None.*

### CODE REVIEW

**Review by ZacButko — APPROVED** [review]
Looks good! Some minor nit-picky things. The only thing I would ask is to translate Gender if that makes sense to you

Thanks Vineet!

File: packages/esm-patient-registration-app/src/widgets/display-photo.component.tsx [hunk 1]
```diff
@@ -2,27 +2,29 @@ import React from 'react';
...
-import styles from './display-photo.scss';
 
 interface DisplayPatientPhotoProps {
   patientName: string;
   patientUuid: string;
+  size?: string;
 }
 
-export default function DisplayPatientPhoto({ patientUuid, patientName }: DisplayPatientPhotoProps) {
+export default function DisplayPatientPhoto({ patientUuid, patientName, size = 'normal' }: DisplayPatientPhotoProps) {
```
[the remark this answers is not in the document]
Comment (vasharma05) [reply packages/esm-patient-registration-app/src/widgets/display-photo.component.tsx:12]: I added for safety, but it has no use.
No thanks!

File: packages/esm-patient-search-app/src/patient-search-result/patient-search-result.component.tsx [hunk 2]
```diff
@@ -1,19 +1,63 @@
 import React, { useCallback, useMemo } from 'react';
 import styles from './patient-search-result.scss';
-import { ExtensionSlot, useConfig, interpolateString, navigate } from '@openmrs/esm-framework';
+import { ExtensionSlot, useConfig, interpolateString, navigate, ConfigurableLink } from '@openmrs/esm-framework';
 import { SearchedPatient } from '../types/index';
+import { SkeletonIcon, SkeletonText } from 'carbon-components-react';
+
+function getGender(gender) {
```
[the remark this answers is not in the document]
Comment (vasharma05) [reply packages/esm-patient-search-app/src/patient-search-result/patient-search-result.component.tsx:7]: Totally forgot

File: packages/esm-patient-search-app/src/patient-search-result/patient-search-result.component.tsx [hunk 3]
```diff
@@ -54,27 +98,60 @@ const PatientSearchResults: React.FC<PatientSearchResultsProps> = ({ patients, h
...
               }}
             />
           </div>
-        </div>
+          <div>
+            <h2 className={styles.patientName}>{`${patient.name?.[0]?.given?.join(' ')} ${
+              patient.name?.[0]?.family
+            }`}</h2>
+            <p className={styles.demographics}>
+              {getGender(patient.gender)} <span className={styles.middot}>&middot;</span> {getAge(patient.birthDate)}{' '}
```
[the remark this answers is not in the document]
Comment (vasharma05) [reply packages/esm-patient-search-app/src/patient-search-result/patient-search-result.component.tsx:123]: Done!
Thanks!

File: packages/esm-patient-search-app/src/patient-search-result/patient-search-result.scss [hunk 4]
```diff
@@ -1,35 +1,32 @@
...
-  color: $text-01;
+  @include carbon--type-style('productive-heading-02');
+  color: $text-02;
 }
 
 .patientAvatar {
-  width: 5rem;
-  height: 5rem;
-  margin: 1rem 1rem 1rem 0;
+  width: 3rem;
```
[the remark this answers is not in the document]
Comment (vasharma05) [reply packages/esm-patient-search-app/src/patient-search-result/patient-search-result.scss:27]: Missed it out.
Thanks!

File: packages/esm-patient-search-app/src/patient-search-result/patient-search-result.scss [hunk 5]
```diff
@@ -56,3 +53,7 @@
   padding-top: 0.5rem;
```
[the remark this answers is not in the document]
Comment (vasharma05) [reply packages/esm-patient-search-app/src/patient-search-result/patient-search-result.scss:53]: Done!

File: packages/esm-patient-search-app/src/patient-search/patient-search.component.tsx [hunk 6]
```diff
@@ -1,91 +1,118 @@
...
+                <br />
+                <span>{t('orPatientName', "OR the patient's name(s)")}</span>
+              </p>
+            </Tile>
+          </div>
+        )
+      ) : (
         <div className={styles.searchResults}>
-          <p className={styles.resultsText}>{t('errorText', 'An error occurred while performing search')}</p>
+          {/* <p className={styles.labelText}>{t('errorText', 'An error occurred while performing search')}</p> */}
```
[the remark this answers is not in the document]
Comment (vasharma05) [reply packages/esm-patient-search-app/src/patient-search/patient-search.component.tsx:107]: Deleted!

**Pull request merged**